src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -207,8 +201,7 @@ private void OnHostingHttpRequestInStart(object arg)
                 return;
             }
 
-            var httpContext = HttpRequestInStartHttpContextFetcher.Fetch<HttpContext>(arg);
-            HttpRequest request = httpContext.Request;
+            HttpRequest request = arg.As<HttpRequestInStartStruct>().HttpContext.Request;
```

Comment:
I think you've already separated these AspNetCoreDiagnosticObserver.cs changes into #1038, so could you revert the changes to these files in this PR?

Reply:
Yes I was planning to do that :)

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/1007/commits/b1a2946d8a03e4a1b27a31797377153ea022b011